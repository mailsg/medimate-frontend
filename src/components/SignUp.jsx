import React, { Component } from 'react';
import axios from 'axios';
import { NavLink } from 'react-router-dom';

class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      username: '',
      email: '',
      password: '',
      confirmPassword: '',
      error: '',
      successMessage: '',
    };
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = async (e) => {
    e.preventDefault();
    const {
      username, email, password, confirmPassword,
    } = this.state; if (password !== confirmPassword) {
      this.setState({ error: 'Passwords do not match' });
      return;
    } try {
      await axios.post('http://localhost:3000/users/', {
        username,
        email,
        password,
      }); // Handle successful registration here
      this.setState({ successMessage: 'Registration successful', error: '' });
    } catch (error) {
      // Handle registration failure
      this.setState({ error: 'Registration failed', successMessage: '' });
    }
  };

  render() {
    const {
      username,
      email,
      password,
      confirmPassword,
      error,
      successMessage,
    } = this.state;
    return (
      <div className="signup-container">
        <h2>Sign Up</h2>
        <form onSubmit={this.handleSubmit}>
          <div>
            <label htmlFor="username">Username</label>
            <input
              type="text"
              id="username"
              name="username"
              value={username}
              onChange={this.handleChange}
              required
            />
          </div>
          <div>
            <label htmlFor="email">Email</label>
            <input
              type="email"
              id="email"
              name="email"
              value={email}
              onChange={this.handleChange}
              required
            />
          </div>
          <div>
            <label htmlFor="password">Password</label>
            <input
              type="password"
              id="password"
              name="password"
              value={password}
              onChange={this.handleChange}
              required
            />
          </div>
          <div>
            <label htmlFor="confirmPassword">Confirm Password</label>
            <input
              type="password"
              id="confirmPassword"
              name="confirmPassword"
              value={confirmPassword}
              onChange={this.handleChange}
              required
            />
          </div>
          <p>Already have an account? Login</p>
          <div>
            <button type="submit">Sign Up</button>
            <NavLink to="/">
              <button type="submit">Login</button>
            </NavLink>
          </div>
          {error && <p className="error-message">{error}</p>}
          {successMessage && (
            <p className="success-message">{successMessage}</p>
          )}
        </form>
      </div>
    );
  }
} export default SignUp;
