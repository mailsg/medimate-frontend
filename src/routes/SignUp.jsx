/* eslint-disable jsx-a11y/label-has-associated-control, camelcase */

import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { registerUser } from '../redux/slice/authSlice'; // Import your register action
import '../App.css';

const SignupPage = () => {
  const dispatch = useDispatch();
  const [formData, setFormData] = useState({
    username: '',
    email: '',
    password: '',
    password_confirmation: '',
  });

  const {
    username, email, password, password_confirmation,
  } = formData;

  const handleChange = (e) => {
    setFormData({ ...formData, [e.target.name]: e.target.value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    // Check if passwords match
    if (password !== password_confirmation) {
      // Handle password mismatch error
      return;
    }

    // Dispatch the registerUser action with the form data
    dispatch(registerUser({ username, email, password }));

    // Reset form fields
    setFormData({
      username: '',
      email: '',
      password: '',
      password_confirmation: '',
    });
  };

  return (
    <div>
      <h2>Signup</h2>
      <form onSubmit={handleSubmit}>
        <div>
          <label htmlFor="username">Username</label>
          <input
            type="text"
            name="user[username]"
            id="username"
            value={username}
            onChange={handleChange}
            required
          />
        </div>
        <div>
          <label htmlFor="email">Email</label>
          <input
            type="email"
            name="user[email]"
            id="email"
            value={email}
            onChange={handleChange}
            required
          />
        </div>
        <div>
          <label htmlFor="password">Password</label>
          <input
            type="password"
            name="user[password]"
            id="password"
            value={password}
            onChange={handleChange}
            required
          />
        </div>
        <div>
          <label htmlFor="password_confirmation">Confirm Password</label>
          <input
            type="password"
            name="password_confirmation"
            id="confirmPassword"
            value={password_confirmation}
            onChange={handleChange}
            required
          />
        </div>
        <button type="submit">Register</button>
      </form>
    </div>
  );
};

export default SignupPage;
