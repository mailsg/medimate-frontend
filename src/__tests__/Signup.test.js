import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import SignUp from '../components/auth_pages/Signup';

describe('Sign up component', () => {
  it('Should render the component correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <SignUp />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
