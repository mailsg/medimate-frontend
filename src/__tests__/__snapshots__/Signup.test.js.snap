// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up component Should render the component correctly 1`] = `
<div>
  <div
    class="reservation-container"
  >
    <h1
      class="reserve-form-header"
    >
      Sign Up
    </h1>
    <form
      class="reservation-form"
    >
      <div
        class="form-group"
      >
        <input
          name="username"
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <input
          name="email"
          placeholder="Email"
          type="email"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <input
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <input
          name="password_confirmation"
          placeholder="Confirm Password"
          type="password"
          value=""
        />
      </div>
      <button
        class="submit-button"
        type="submit"
      >
        Sign Up
      </button>
    </form>
  </div>
</div>
`;
